driver-core/src/main/com/mongodb/internal/async/AsyncRunnable.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.async;
+
+import com.mongodb.internal.async.function.RetryState;
+import com.mongodb.internal.async.function.RetryingAsyncCallbackSupplier;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * See tests for usage (AsyncFunctionsTest).
```

Comment:
I prefer that usage documentation is placed in the code rather than the test.  It will be easier for maintainers to refer to it if it's here.  And that's where all of our existing internal Javadoc is, so it will be less surprising.

Of all the new interfaces, this seems like the best place to put it.

Reply:
Moved; I find that I often look at the test code (more often than the docs) to confirm that I am using the API correctly.